src/scorekeeper.ts

```diff
@@ -101,6 +101,30 @@ export default class ScoreKeeper {
       }
     );
 
+    // Handles offline event. Validators will be faulted for each session they are offline.
+    //     If they have already reaceived an offline fault for that session, it is skipped
+    this.handler.on("someOffline", async (data: { offlineVals: string[] }) => {
```

Comment:
Isn't someOffline an ending session event ? Therefore
const session = await this.chaindata.getSession();
will be equals to sessionWithFaults+1, isn't it ?